linuxrc.c

```diff
@@ -1148,53 +1147,75 @@ void lxrc_init()
   if(config.braille.check) run_braille();
 
   if(!config.manual && !auto2_init()) {
+    char *buf = NULL, *repo = NULL;
+
     fprintf(stderr, "Automatic setup not possible.\n");
 
-    util_disp_init();
+    // ok, we failed to find a suitable repo
+    // do something about it
 
-    i = 0;
-    j = 1;
+    // If the file '/nextmedia' exists, get the message to show from there
+    // and retry the default repo settings if the user is ready.
+    // Otherwise, present the user the list of repos we have tried so far and
+    // let her choose / edit one and try again.
+
+    util_disp_init();
 
     if(util_check_exist("/nextmedia") == 'r') {
       config.cd1texts = file_parse_xmllike("/nextmedia", "text");  
     }
 
-    if(config.url.install) {
-      char *s = get_translation(config.cd1texts, current_language()->locale);
-      char *buf = NULL;
+    char *next_msg = get_translation(config.cd1texts, current_language()->locale);
+
+    strprintf(&buf,
+      "%s\n\n%s",
+      next_msg ?: "Please make sure your installation medium is available.",
+      config.cd1texts ? "Continue?" : "Choose the URL to retry."
+    );
 
-      strprintf(&buf, "%s\n\nRetry?", s ?: "Please make sure your installation medium is available.");
-      do {
-        j = dia_yesno(buf, YES) == YES ? 1 : 0;
-        if(j) {
-          slist_t *sl;
+    do {
+      config.manual = 0;
+
+      if(config.cd1texts) {
```

Comment:
work-flow in this if statement is a bit confusing as it actually do not use look from do ( always break at the end ).
So maybe it deserve own method and work-flow like:

```
config.manual = 0;
if(config.cd1texts) {
  ....
} else {
  do {
     ...
  }
}
```
of course unless I overlook that else branch can set `config.cd1texts`,  but from current code it doesn't look like.